### PR TITLE
Added psycopg3 support

### LINKED ISSUE #1736: Add Support for psycopg3
With the release of django 4.2a we can use psycopg3. However [django-debug-toolbar](https://github.com/jazzband/django-debug-toolbar) is not compatible with it.

### PR DESCRIPTION
**Pull request by nofalx**
# Description

Add support for psycopg3 while still keeping support for psycopg2

Fixes #1736

# Checklist:

- [x] I have added the relevant tests for this change.
- [x] I have added an item to the Pending section of ``docs/changes.rst``.
- [x] Fix test `tests.panels.test_sql.SQLPanelMultiDBTestCase.test_transaction_status`

### CODE REVIEW

**Comment by nofalx** [comment]
@tim-schilling please add your changes for `test_tuple_param_conversion` to this PR instead

**Comment by tim-schilling** [comment]
Sorry, I dropped the ball on this.

**Comment by tim-schilling** [comment]
This is getting closer. I'm struggling to get `tox-gh-actions` to work properly. I might end up ripping it out

**Comment by tim-schilling** [comment]
It turns out we were never running the postgis test suite. That's been fixed.
If all the tests pass, I'll clean up the PR and get it merged in.

**Comment by tim-schilling** [comment]
Phew, that is done.

**Review by tim-schilling — APPROVED** [review]

**Comment by tim-schilling** [comment]
@matthiask can you double check the changes for me?

**Comment by YAmikep** [comment]
For what it's worth, I just tested it with the following and it works:
- Python 3.11.2
- psycopg==3.1.8
- PostgreSQL 15.2

Thanks you 🙌

File: docs/changes.rst [hunk 1]
```diff
@@ -14,6 +14,7 @@ Pending
   memory leaks and sometimes very verbose and hard to silence output in some
   environments (but not others). The maintainers judged that time and effort is
   better invested elsewhere.
+* Added support for psycopg3 3.
```
Comment (matthiask) [inline docs/changes.rst:17]:
```suggestion
* Added support for psycopg3 3.
```

I'm not sure what's better, but this should probably be either "psycopg3" or "psycopg 3", not "psycopg3 3"

**Review by matthiask — APPROVED** [review]

**Review by islam-kamel — APPROVED** [review]

**Review by islam-kamel — APPROVED** [review]

File: docs/changes.rst (hunk 1 above)
Comment (tim-schilling) [reply docs/changes.rst]: That one is on me. I changed it to get past the spellcheck originally. I must not have cleaned up after myself properly.

**Pull request merged**